### PR TITLE
[NLU-2621] Add datetime recognizer for PT

### PR DESCRIPTION
**Pull request by samhickey25**

### CODE REVIEW

**Review by ushmau5 — APPROVED** [review]

**Review by andrew-gradinari — APPROVED** [review]

File: Python/libraries/recognizers-number-with-unit/recognizers_number_with_unit/resources/portuguese_numeric_with_unit.py [hunk 1]
```diff
@@ -599,7 +599,7 @@ class PortugueseNumericWithUnit:
                                         ("Ngwee", "NGWEE"),
                                         ("Millibitcoin", "MILLIBITCOIN"),
                                         ("Satoshi", "SATOSHI")])
-    CompoundUnitConnectorRegex = f'(?<spacer>e|com)'
+    CompoundUnitConnectorRegex = f'\\b(?<spacer>e|com)\\b'
```
Comment (samhickey25) [inline Python/libraries/recognizers-number-with-unit/recognizers_number_with_unit/resources/portuguese_numeric_with_unit.py:602]: This isn't being picked up in the built package version for some reason and is causing issues

File: Python/tests/runner.py [hunk 2]
```diff
@@ -72,17 +72,17 @@ def get_specs(recognizer, entity):
...
+    # 'Italian': Culture.Italian,
+    # 'Japanese': Culture.Japanese,
+    # 'Korean': Culture.Korean,
     'Portuguese': Culture.Portuguese,
-    'Spanish': Culture.Spanish,
-    'Turkish': Culture.Turkish,
-    'German': Culture.German,
+    # 'Spanish': Culture.Spanish,
+    # 'Turkish': Culture.Turkish,
+    # 'German': Culture.German,
```
Comment (andrew-gradinari) [inline Python/tests/runner.py:85]: Please don't forget to uncomment the tests.

**Review by kevinwalshgen — APPROVED** [review]

**Comment by colm-dillon** [comment]
Hey Sam should we be modifying the files like we are here? 

@sanhal can you weigh in here

See below. Some of the files are auto generated - and could be the reason you're not seeing it working functioanally


# ------------------------------------------------------------------------------
# <auto-generated>
#     This code was generated by a tool.
#     Changes to this file may cause incorrect behavior and will be lost if
#     the code is regenerated.
# </auto-generated>
#
# Copyright (c) Microsoft Corporation. All rights reserved.
# Licensed under the MIT License.
# ------------------------------------------------------------------------------

**Comment by samhickey25** [comment]
> 

Ah I see the issue. In the jenkinsfile we build the .yaml resources, and not the python resources

**Review by colm-dillon — APPROVED** [review]

File: Patterns/Portuguese/Portuguese-DateTime.yaml [hunk 3]
```diff
@@ -935,7 +935,7 @@ YearSuffix: !nestedRegex
   references: [ YearRegex, FullTextYearRegex ]
 SuffixAfterRegex: !simpleRegex
   # TODO: modify below regex according to the counterpart in English
-  def: ^[.]
+  def: ^\b$
```
Comment (sanhal) [inline Patterns/Portuguese/Portuguese-DateTime.yaml:938]: Could you explain this change?

Comment (samhickey25) [reply Patterns/Portuguese/Portuguese-DateTime.yaml:938]: Yeah so for the example "11:55." The period was getting picked up in the extractor and then nothing would be resolved. 
I compared this with other languages and matches and found that this was the required regex

**Review by sanhal — APPROVED** [review]

**Review by andrew-gradinari — APPROVED** [review]

**Pull request merged**